native/src/main/java/io/ballerina/stdlib/data/xml/XmlParser.java

```diff
@@ -0,0 +1,853 @@
+            }
+        }
+        return attributeName;
+    }
+
+    private void handleAttributes(XMLStreamReader xmlStreamReader, XmlParserData xmlParserData) {
+        for (int i = 0; i < xmlStreamReader.getAttributeCount(); i++) {
+            QName attributeQName = xmlStreamReader.getAttributeName(i);
+            QualifiedName attQName = new QualifiedName(attributeQName.getNamespaceURI(),
+                    attributeQName.getLocalPart(), attributeQName.getPrefix());
+            Field field = xmlParserData.attributeHierarchy.peek().remove(attQName);
+            if (field == null) {
+                field = xmlParserData.fieldHierarchy.peek().get(attQName);
+            }
+
+            if (field == null) {
+                return;
+            }
+
+            try {
+                currentNode.put(StringUtils.fromString(field.getFieldName()), convertStringToExpType(
+                        StringUtils.fromString(xmlStreamReader.getAttributeValue(i)), field.getFieldType()));
+            } catch (Exception e) {
+                // Ignore: Expected type will mismatch when element and attribute having same name.
+            }
+        }
+    }
+
+    private void handleAttributesRest(XMLStreamReader xmlStreamReader, Type restType, BMap<BString, Object> mapNode) {
+        for (int i = 0; i < xmlStreamReader.getAttributeCount(); i++) {
+            QName attributeQName = xmlStreamReader.getAttributeName(i);
+            QualifiedName attQName = new QualifiedName(attributeQName.getNamespaceURI(),
+                    attributeQName.getLocalPart(), attributeQName.getPrefix());
+
+            try {
+                mapNode.put(StringUtils.fromString(attQName.getLocalPart()), convertStringToRestExpType(
+                        StringUtils.fromString(xmlStreamReader.getAttributeValue(i)), restType));
+            } catch (Exception e) {
+                // Ignore: Expected type will mismatch when element and attribute having same name.
+            }
+        }
+    }
+
+    private Optional<Object> handleRecordRestType(XmlParserData xmlParserData, XMLStreamReader xmlStreamReader) {
+        xmlParserData.currentField = null;
+        Type restType = TypeUtils.getReferredType(xmlParserData.restTypes.peek());
+        int restTypeTag = restType.getTag();
+        QualifiedName elementQName = getElementName(xmlStreamReader);
+        String fieldName = elementQName.getLocalPart(); // Since local part equals to field name.
+        if (restTypeTag == TypeTags.RECORD_TYPE_TAG) {
+            RecordType recordType = (RecordType) restType;
+            currentNode = updateNextValue(recordType, fieldName, restType, xmlParserData);
+            handleAttributes(xmlStreamReader, xmlParserData);
+            parseRecordRest(fieldName, xmlParserData);
+            xmlParserData.siblings.clear();
+            return Optional.ofNullable(xmlParserData.nodesStack.pop());
+        } else if (restTypeTag == TypeTags.ARRAY_TAG) {
```

Comment:
```suggestion
        } if (restTypeTag == TypeTags.ARRAY_TAG) {
```

Reply:
Since those are relevant. It is better to keep else-if since it will improve the understandability of the code